### PR TITLE
[SandboxVec][Interval] Convert InstrInterval class to a class template

### PR DESCRIPTION
**Pull request by vporpo**
This patch converts InstrInterval class to a class template and renames InstrInterval to Itnerval.

This change will allow us to reuse the Interval for dependency graph nodes.

### CODE REVIEW

**Comment by vporpo** [comment]
Fixed the clang-format issue.

**Comment by vporpo** [comment]
Rebase

File: llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Interval.h [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  T *To;
+
+public:
+  Interval() : From(nullptr), To(nullptr) {}
+  Interval(T *From, T *To) : From(From), To(To) {
+    assert((From == To || From->comesBefore(To)) &&
+           "From should come before TopI!");
+  }
+  Interval(ArrayRef<T *> Instrs) {
+    assert(!Instrs.empty() && "Expected non-empty Instrs!");
```
Comment (tmsri) [inline llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Interval.h:86]: Shouldn't Instrs be re-named here?  Like Say Items?

**Review by tmsri — APPROVED** [review]

File: llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Interval.h (hunk 1 above)
Comment (vporpo) [reply llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Interval.h:86]: Good catch!

File: llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Interval.h [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+template <typename T> class Interval {
+  T *From;
+  T *To;
+
+public:
+  Interval() : From(nullptr), To(nullptr) {}
+  Interval(T *From, T *To) : From(From), To(To) {
+    assert((From == To || From->comesBefore(To)) &&
+           "From should come before TopI!");
```
[the remark this answers is not in the document]
Comment (vporpo) [reply llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Interval.h:83]: Done. This should be `From`, left over from a previous refactoring.

**Pull request merged**